src/Dont/Exception/NonDeSerializableObject.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Dont\Exception;
+
+use LogicException;
+
+class NonDeSerializableObject extends LogicException implements ExceptionInterface
+{
+    public static function fromAttemptedDeSerialization($object) : self
+    {
+        if (! is_object($object)) {
+            throw TypeError::fromNonObject($object);
+        }
+
+        $template = <<<'ERROR'
```

Comment:
I get the purpose, but these messages are *really* verbose... Surely a simple `%s objects should not be (de)serialized` would suffice? Make the point, but in less characters basically...